OWL2/java/de/unibremen/informatik/OWL2Parser.java

```diff
@@ -402,32 +405,92 @@ void renderAsOmn(OWLOntology onto) {
 		}
 
 		void renderAsXml(OWLOntology onto) {
+			FileOutputStream fos = null;
+			FileInputStream fis = null;
+			OutputStreamWriter out = null;
+			InputStreamReader in = null;
+			char[] buf = new char[8192];
+			int c, n;
+			int chomp = -1;
 			try {
 				OWLXMLRenderer xmlren = new OWLXMLRenderer();
-				File tempFile = File.createTempFile("owlTemp_1", ".xml");
-				FileWriter buf = new FileWriter(tempFile);
+				String t = System.getenv("HETS_GZIP");
+				boolean gz = t == null || t.equalsIgnoreCase("on");
+				t = gz ? ".xml.gz" : ".xml";
```

Comment:
please use speaking variable names, and do not re-use variable names. Even if in this case, this is not dramatic, we should have this as a general guideline.